tests/integration/s2n_handshake_test_s_client.py

```diff
@@ -325,30 +340,30 @@ def elliptic_curve_test(host, port):
             # Put some unsupported curves in front to make sure we gracefully skip them
             mixed_curves = unsupported_curves + list(permutation)
             mixed_curves_str = ':'.join(mixed_curves)
-            for cipher in filter(lambda x: x.openssl_name == "ECDHE-RSA-AES128-GCM-SHA256" or x.openssl_name == "ECDHE-RSA-AES128-SHA", ALL_TEST_CIPHERS):
-                ret = try_handshake(host, port, cipher.openssl_name, S2N_TLS12, curves=mixed_curves_str)
+            for cipher in filter(lambda x: x.openssl_name == "ECDHE-RSA-AES128-GCM-SHA256" or x.openssl_name == "ECDHE-RSA-AES128-SHA256", ALL_TEST_CIPHERS):
```

Comment:
I'm assuming that SHA256 works in FIPS mode, but SHA doesn't? Should we test with all 3 ciphersuite names in non-FIPS mode and only filter out SHA when in FIPS mode?